subsys/net/lib/nrf_cloud/src/nrf_cloud_pgps.c

```diff
@@ -1615,21 +1683,30 @@ int nrf_cloud_pgps_begin_update(void)
 
 int nrf_cloud_pgps_finish_update(void)
 {
+#if defined(CONFIG_PM_PARTITION_REGION_PGPS_EXTERNAL)
+	if (state != PGPS_READY) {
```

Comment:
Forgive me for not having a solid grasp of this module's state transitions, but, If PGPS lookup starts right as download finishes, is it possible this could end up accidentally unlocking the lock that PGPS lookup grabbed?

Reply:
State is set to PGPS_READY when the download finishes without error, in which case, this block will not give the semaphore.

Reply:
I do want to clean this up though.  I added `nrf_cloud_pgps_finish_update` and its begin partner, as well as the non-HTTP download transport feature, so that LWM2M can be the transport.  I need to rewrite parts of this so the finish update call can be made for either download transport, so the end of a transfer (error or no error) always calls this, and gives the semaphore.